modules/core/src/main/scala/gem/parser/Angle.scala

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package parser
+
+import atto._, Atto._
+import cats.implicits._
+import gem.math.{ Angle, HourAngle }
+
+/** Parsers for `[[gem.math.Angle]].` */
+trait AngleParsers {
+  import MiscParsers._
+
+  /**
+   * Generic parser for the components of an angle in "11 22 33.444555" format, with at least 1
+   * and at most 6 digits following the decimal point, and terminal parsers for each segment.
+   */
+  def genAngle(t1: Parser[_], t2: Parser[_], t3: Parser[_]): Parser[(Int, Int, Int, Int, Int)] =
+    (int <~ t1, int <~ t2, int <~ char('.'), frac(6) <~ t3) mapN { (h, m, s, µs) =>
+      (h, m, s, µs / 1000, µs % 1000)
+    } named s"genAngle($t2, $t2, $t3)"
+
+  /** Generic parser for the components of an HourAngle; see `genAngle`. */
+  def genHMS(t1: Parser[_], t2: Parser[_], t3: Parser[_]): Parser[HourAngle] =
+    genAngle(t1, t2, t3).map((HourAngle.fromHMS _).tupled) named s"genHMS($t2, $t2, $t3)"
```

Comment:
Why `genHMS($t2, $t2, $t3)` and not `genHMS($t1, $t2, $t3)`?
(Same comment for `genAngle`.)